packages/@aws-cdk/aws-ec2/lib/vpc-endpoint.ts

```diff
@@ -0,0 +1,425 @@
+import iam = require('@aws-cdk/aws-iam');
+import cdk = require('@aws-cdk/cdk');
+import { Connections, IConnectable } from './connections';
+import { CfnVPCEndpoint } from './ec2.generated';
+import { SecurityGroup } from './security-group';
+import { VpcSubnet } from './vpc';
+import { IVpcNetwork, IVpcSubnet, SubnetSelection } from './vpc-ref';
+
+/**
+ * A VPC endpoint.
+ */
+export interface IVpcEndpoint extends cdk.IConstruct {
+  /**
+   * The VPC endpoint identifier.
+   */
+  readonly vpcEndpointId: string;
+}
+
+/**
+ * A VPC gateway endpoint.
+ */
+export interface IVpcGatewayEndpoint extends IVpcEndpoint {
+  /**
+   * Exports this VPC endpoint from the stack.
+   */
+  export(): VpcGatewayEndpointImportProps;
+}
+
+/**
+ * The type of VPC endpoint.
+ */
+export enum VpcEndpointType {
+  /**
+   * Interface
+   */
+  Interface = 'Interface',
+
+  /**
+   * Gateway
+   */
+  Gateway = 'Gateway'
+}
+
+/**
+ * A VPC endpoint service.
+ */
+export class VpcEndpointService {
+  constructor(public readonly name: string, public readonly type: VpcEndpointType) {}
+}
+
+/**
+ * A VPC endpoint AWS service.
```

Comment:
Would say *A VPC endpoint for an AWS service*

Reply:
_An AWS VPC endpoint service_? This is a service not an endpoint...

Reply:
I kinda thing of those as "services that provide VPC endpoints"... But maybe my mental model is wrong here... This is actually why I was calling for @rix0rrr  to have a look, too... I feel he might be a little more into this domain than I am.